src/redux/actions/tripsActions/approveRequestAction.js

```diff
@@ -0,0 +1,34 @@
+import { toast } from 'react-toastify';
+import { SPINNER_STATUS, APPROVE_REQUEST_SUCCESS, APPROVE_REQUEST_ERROR } from '../../actionTypes/tripsActionTypes';
+import apiCall from '../../../helpers/apiCall';
+
+export const updateSpinnerStatus = (value) => ({
+  type: SPINNER_STATUS,
+  spinner: value,
+});
+
+const approveRequest = (tripRequestId, query) => async (dispatch) => {
+  const API_URL = `/trips/${tripRequestId}?${query}`;
+  const HEADERS_REQUEST = {
+    token: localStorage.getItem('token'),
+  };
+  try {
+    const approveRequestData = await apiCall.patch(API_URL, { reason: 'good' }, { headers: HEADERS_REQUEST });
+    dispatch(updateSpinnerStatus(false));
```

Comment:
This `reason` data, I would suggest to get it from user inputs as the backend works Or leave it and use the `comment` field as it is all the same.

Reply:
@william000000 I chose to go with the comment field otherwise what would be the reason for the comment feature?